frontend/src/pages/CompareV2.tsx

```diff
@@ -51,22 +109,81 @@ function CompareV2(props: PageProps) {
   const runIds = (queryParamRunIds && queryParamRunIds.split(',')) || [];
 
   // Retrieves run details.
-  const { isError, data, refetch } = useQuery<ApiRunDetail[], Error>(
+  const {
+    isLoading: isLoadingRunDetails,
+    isError: isErrorRunDetails,
+    error: errorRunDetails,
+    data: runs,
+    refetch,
+  } = useQuery<ApiRunDetail[], Error>(
     ['run_details', { ids: runIds }],
     () => Promise.all(runIds.map(async id => await Apis.runServiceApi.getRun(id))),
     {
       staleTime: Infinity,
-      onError: async error => {
-        const errorMessage = await errorToMessage(error);
+    },
+  );
+
+  // Retrieves MLMD states (executions and linked artifacts) from the MLMD store.
+  const {
+    data: mlmdPackages,
+    isLoading: isLoadingMlmdPackages,
+    isError: isErrorMlmdPackages,
+    error: errorMlmdPackages,
+  } = useQuery<MlmdPackage[], Error>(
+    ['run_artifacts', { runIds }],
+    () =>
+      Promise.all(
+        runIds.map(async runId => {
+          const context = await getKfpV2RunContext(runId);
+          const executions = await getExecutionsFromContext(context);
+          const artifacts = await getArtifactsFromContext(context);
+          const events = await getEventsByExecutions(executions);
+          return {
+            executions,
+            artifacts,
+            events,
+          } as MlmdPackage;
+        }),
+      ),
+    {
+      staleTime: Infinity,
+    },
+  );
+
+  const runArtifacts = runs && mlmdPackages ? getRunArtifacts(runs, mlmdPackages) : [];
```

Comment:
NIT: No need to make a change here because you are going to use `runArtifacts` very soon. As a general practice: If you are not going to use the return value of a function, you can omit the `const runArtifacts =`, and add a TODO to explain the pending work item will need `getRunArtifacts`.